datashader/glyphs.py

```diff
@@ -444,6 +570,33 @@ def extend_line(vt, bounds, xs, ys, plot_start, *aggs_and_cols):
     return extend_line
 
 
+def _build_extend_lines_xy(draw_line, map_onto_pixel):
+    extend_line = _build_extend_line(draw_line, map_onto_pixel)
+
+    @ngjit
+    def extend_lines_xy(vt, bounds, xs, ys, plot_start, *aggs_and_cols):
```

Comment:
I've had a go at rewriting this by adapting the ``extend_line`` code and that seems somewhat more efficient than unpacking this format and reusing the existing codepath. Probably needs some more fixes, but here's what my 10 minutes of hacking produced:

```

    @ngjit
    def extend_lines_xy(vt, bounds, xs, ys, plot_start, *aggs_and_cols):
        """
        here xs and ys are tuples of arrays and non-empty
        """
        xmin, xmax, ymin, ymax = bounds
        nrows = xs[0].shape[0]
        ncols = len(xs)
        i = 0
        while i < nrows - 1:
            plot_start = True
            j = 0
            while j < ncols - 1:
                x0 = xs[j][i]
                y0 = ys[j][i]
                x1 = xs[j+1][i]
                y1 = ys[j+1][i]

                # Use Liang-Barsky (1992) to clip the segment to a bounding box
                if outside_bounds(x0, y0, x1, y1, xmin, xmax, ymin, ymax):
                    plot_start = True
                    j += 1
                    continue

                t0, t1 = 0, 1
                dx = x1 - x0

                t0, t1, accept = clipt(-dx, x0 - xmin, t0, t1)
                if not accept:
                    j += 1
                    continue

                t0, t1, accept = clipt(dx, xmax - x0, t0, t1)
                if not accept:
                    j += 1
                    continue

                dy = y1 - y0

                t0, t1, accept = clipt(-dy, y0 - ymin, t0, t1)
                if not accept:
                    j += 1
                    continue

                t0, t1, accept = clipt(dy, ymax - y0, t0, t1)
                if not accept:
                    j += 1
                    continue

                if t1 < 1:
                    clipped = True
                    x1 = x0 + t1 * dx
                    y1 = y0 + t1 * dy

                if t0 > 0:
                    # If x0 is clipped, we need to plot the new start
                    clipped = True
                    plot_start = True
                    x0 = x0 + t0 * dx
                    y0 = y0 + t0 * dy

                x0i, y0i = map_onto_pixel(vt, bounds, x0, y0)
                x1i, y1i = map_onto_pixel(vt, bounds, x1, y1)
                draw_line(x0i, y0i, x1i, y1i, i, plot_start, clipped, *aggs_and_cols)
                plot_start = False
                j += 1
            i += 1

    return extend_lines_xy
```

Reply:
Cool!  @jonmmease, can you benchmark that like you did above?

Reply:
Using this on a 10M vertex trimesh wireframe using dask this renders in 0.5 seconds as opposed to 1.5 seconds with ``cvs.line``.

Reply:
Awesome, thanks for giving this a try @philippjfr! I'll incorporate this approach and add it to my benchmarks